### PR TITLE
Improved test coverage

### PR DESCRIPTION
**Pull request by Insti**
Here are a bunch of changes that improve the test coverage.

Improved code coverage, the only things NOT tested now are: 
* non-"test" environment variable configuration.
* The custom 500 error handling. (I still need to work out how to trigger this from within the test environment.)

Test environment changes:
Tests can now be run individually and have their coverage results combined.
The test files are now excluded from the coverage report.

(Note these are not squashed as they should all be independent.)

### CODE REVIEW

**Comment by Insti** [comment]
Note Coveralls shows a decrease in overall coverage due to the removal of all the "100% covered" test files, but the number of untested lines has decreased.

**Comment by kytrinyx** [comment]
> The test files are now excluded from the coverage report.

Yay! This is so much better.

File: lib/xapi/tracks.rb [hunk 1]
```diff
@@ -18,11 +18,6 @@ def [](id)
       by_id[id]
     end
 
-    # rubocop:disable Style/OpMethod
-    def -(ids)
```
Comment (kytrinyx) [inline lib/xapi/tracks.rb]: Was this not being used?

**Comment by kytrinyx** [comment]
Thanks, @Insti!

File: lib/xapi/tracks.rb (hunk 1 above)
Comment (kytrinyx) [inline lib/xapi/tracks.rb]: Oh, right. Saw the separate commit.

**Pull request merged**